frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetailWrapper.tsx

```diff
@@ -43,6 +43,10 @@ export function ObjectDetailWrapper({
 
   const hasPagination = data?.rows?.length > 1;
 
+  if (data.rows.length <= currentObjectIndex) {
+    setCurrentObjectIndex(data.rows.length - 1);
```

Comment:
:x: `setState` should not be called on every render. That's what effects are for.